library/Booster/Syntax/ParsedKore/Internalise.hs

```diff
@@ -772,6 +804,10 @@ internaliseRewriteRuleNoAlias partialDefinition exs left right axAttributes = do
                     [ UndefinedSymbol s.name
                     | s <- Util.filterTermSymbols (not . Util.isDefinedSymbol) rhs.term
                     ]
+                        <> [ UndefinedSymbol s.name
+                           | c <- Set.toList lhs.constraints
+                           , s <- Util.filterTermSymbols (not . Util.isDefinedSymbol) $ coerce c
+                           ]
```

Comment:
We should include the partial symbols check of the requires clause, especially now that we are sending stuff to the SMT solver. I think it's sound to do so, only if for every partial symbol `f(...)` we first check whether `#Ceil ( f(...) )` holds.